libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs

```diff
@@ -6,89 +6,107 @@
-        /// <param name="luisRecognizerOptions">The LUIS recognizer options to use.</param>
-        /// <param name="options">The LUIS request options to use.</param>
-        public LuisRecognizer(ILuisModel luisModel, ILuisRecognizerOptions luisRecognizerOptions = null, ILuisOptions options = null)
-        {
-            _luisService = new LuisService(luisModel);
-            _luisOptions = options ?? new LuisRequest();
-            _luisRecognizerOptions = luisRecognizerOptions ?? new LuisRecognizerOptions { Verbose = true };
-        }
-
-        /// <inheritdoc />
-        public async Task<RecognizerResult> RecognizeAsync(string utterance, CancellationToken ct)
+        /// <param name="application">The LUIS application to use to recognize text.</param>
+        /// <param name="predictionOptions">The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">TRUE to include raw LUIS API response.</param>
+        public LuisRecognizer(LuisApplication application, LuisPredictionOptions predictionOptions = null, bool includeApiResults = false)
         {
-            return await RecognizeInternalAsync(utterance, ct).ConfigureAwait(false);
+            runtime = new LuisRuntimeAPI(new ApiKeyServiceClientCredentials(application.EndpointKey))
+            {
+                AzureRegion = (AzureRegions)Enum.Parse(typeof(AzureRegions), application.AzureRegion),
+            };
+            this.application = application;
+            this.options = predictionOptions ?? new LuisPredictionOptions();
+            this.includeApiResults = includeApiResults;
         }
 
         /// <inheritdoc />
-        public async Task<T> RecognizeAsync<T>(string utterance, CancellationToken ct)
+        public async Task<RecognizerResult> RecognizeAsync(ITurnContext context, CancellationToken ct)
+            => await RecognizeInternalAsync(context, ct).ConfigureAwait(false);
+
+        /// <inheritdoc />
+        public async Task<T> RecognizeAsync<T>(ITurnContext context, CancellationToken ct)
             where T : IRecognizerConvert, new()
         {
             var result = new T();
-            result.Convert(await RecognizeInternalAsync(utterance, ct).ConfigureAwait(false));
+            result.Convert(await RecognizeInternalAsync(context, ct).ConfigureAwait(false));
             return result;
-        }
-
-        private static string NormalizedIntent(string intent)
-        {
-            return intent.Replace('.', '_').Replace(' ', '_');
         }
 
-        private static JObject GetIntents(LuisResult luisResult)
-        {
-            return luisResult.Intents != null ?
-                JObject.FromObject(luisResult.Intents.ToDictionary(
+        private static string NormalizedIntent(string intent) => intent.Replace('.', '_').Replace(' ', '_');
+
+        private static IDictionary<string, IntentScore> GetIntents(LuisResult luisResult)
+            => luisResult.Intents != null ?
```

Comment:
Consider decomposing this code into simpler components so it can be more easlier tested.